### PR TITLE
Save the current simulation

### PR DESCRIPTION
**Pull request by magopian**
cf https://www.notion.so/Pr-server-la-simulation-en-cours-lors-de-la-navigation-au-sein-du-site-8eb4fc52098a43a8b3cbf5d73e9b1d37

### CODE REVIEW

**Comment by magopian** [comment]
Point d'attention sur l'interaction avec le chargement d'une simulation (url de partage), et le bouton "réinitialiser le simulateur".

File: src/Page/Simulator.elm [hunk 1]
```diff
@@ -399,7 +403,11 @@ shareLinkView session { impact, funit } simulator =
 
 
 saveLinkView : Session -> Model -> Html Msg
-saveLinkView ({ store } as session) ({ query, simulationName } as model) =
+saveLinkView ({ store } as session) ({ simulationName } as model) =
+    let
+        query =
+            store.query
+    in
```
[the remark this answers is not in the document]
Comment (magopian) [reply src/Page/Simulator.elm:410]: Just to keep the code using `query` as is, but I'll change it to use `store.query` instead 👍

File: src/Data/Session.elm [hunk 2]
```diff
@@ -94,6 +98,7 @@ encodeStore : Store -> Encode.Value
 encodeStore store =
     Encode.object
         [ ( "savedSimulations", Encode.list encodeSavedSimulation store.savedSimulations )
+        , ( "query", Inputs.encodeQuery store.query )
```
[the remark this answers is not in the document]
Comment (n1k0) [reply src/Data/Session.elm:101]: I have a feeling that reopening wikicarbone 3 weeks after having played around with a simulation might be surprising if you render it from localStorage… Apart from browser or OS crash, I don't see a strong use case for reopening a simulation from localStorage, but that may just be me. As always, let's ask users and/or wait for them to actually ask for it.

[the remark this answers is not in the document]
Comment (magopian) [reply src/Data/Session.elm:101]: you're right. I'll move it to `Session` straight away (it's not a big code change anyways if we want it after all).

**Pull request merged**